src/decorators.ts

```diff
@@ -145,3 +145,26 @@ function _query(
     });
   };
 }
+
+/**
+ * A TypeScript property decorator factory that causes the decorated method to
+ * be called when a imperative event is fired on the targeted element. `target`
+ * can be either a single element by id, window, or document.
+ *
+ * https://www.polymer-project.org/2.0/docs/devguide/events#imperative-listeners
+ *
+ * @param eventName A string representing the event type to listen for
+ * @param target A single element by id, window, or document to add the listener
+ * to.
+ *
+ */
+export function listen(eventName: string, target: string|Window|Document) {
```

Comment:
I think the type for `target` should probably be just be `string|EventTarget`. It might be weird to specify a Node other than `window` or `document`, but it's possible.